grails-app/domain/org/transmartproject/db/dataquery/highdim/tworegion/DeTwoRegionJunction.groovy

```diff
@@ -0,0 +1,71 @@
+package org.transmartproject.db.dataquery.highdim.tworegion
+
+import groovy.transform.EqualsAndHashCode
+import org.transmartproject.core.dataquery.highdim.tworegion.Junction
+import org.transmartproject.db.dataquery.highdim.DeSubjectSampleMapping
+
+/**
+ * Created by j.hudecek on 4-12-2014.
+ */
+@EqualsAndHashCode()
+class DeTwoRegionJunction implements Serializable, Junction {
+
+    Long upEnd
+    String upChromosome
+    Long upPos
+    Character upStrand
+    Long downEnd
+    String downChromosome
+    Long downPos
+    Character downStrand
+    Boolean isInFrame
+    Long id
+    DeSubjectSampleMapping assay
+
+    Boolean isInFrame() {
+        return isInFrame
+    }
+
+    static hasMany = [junctionEvents: DeTwoRegionJunctionEvent]
+
+    static constraints = {
+        junctionEvents(nullable: true)
+    }
+
+    static mapping = {
+        table schema: 'deapp', name: 'de_two_region_junction'
+        version false
+        id column: 'two_region_junction_id'
+
+        upEnd column: 'up_end'
+        upChromosome column: 'up_chr'
+        upPos column: 'up_pos'
+        upStrand column: 'up_strand'
+        downEnd column: 'down_end'
+        downChromosome column: 'down_chr'
+        downPos column: 'down_pos'
+        downStrand column: 'down_strand'
+        isInFrame column: 'is_in_frame'
```

Comment:
Most are redundant (except for `upChromosome`, `downChromosome` -- and `inFrame` after the proposed changes).